### PR TITLE
shaka-packager 3.4.0 (new cask)

### PR DESCRIPTION
**Pull request by garnajee**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [X] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [X] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

File: Casks/s/shaka-packager.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  name "Shaka Packager"
+  desc "Packager for DASH and HLS streaming"
+  homepage "https://github.com/shaka-project/shaka-packager"
+
+  livecheck do
+    url "https://github.com/shaka-project/shaka-packager/releases"
+    strategy :page_match
+    regex(%r{href=.*?/v?(\d+(?:\.\d+)+)/packager-osx-arm64}i)
+  end
+
```
Comment (daeho-ro) [inline Casks/s/shaka-packager.rb:15]:
```suggestion
  livecheck do
    url :url
    strategy :github_latest
  end

```

File: Casks/s/shaka-packager.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  binary "packager-osx-arm64", target: "packager"
+
+  caveats <<~EOS
```
Comment (daeho-ro) [inline Casks/s/shaka-packager.rb:18]: It seems that you only install binary, so this comment required.
```suggestion

  # No zap stanza required

  caveats <<~EOS
```

**Review by daeho-ro — COMMENTED** [review]
I left some comment.
I saw that there are intel version too, so you may add it with arch stanza.

**Comment by garnajee** [comment]
Thanks for your feedback! Regarding the `sha256` line, since we now have both Intel and ARM support for the `shaka-packager` binary, I'm not entirely sure how to handle this part. Do we need to include two separate `sha256` checksums? If so, how?

I hope this modification works for you, but feel free to let me know if you have any suggestions on how to handle this.

**Comment by daeho-ro** [comment]
`caveats` is allowed only for the Homebrew specfic one, and yours are quite general so I deleted and squash your commits.

**Comment by daeho-ro** [comment]
Hm... signing problem only for intel. Is this can be fixed? Or should we go only arm?

**Comment by garnajee** [comment]
> Hm... signing problem only for intel. Is this can be fixed? Or should we go only with arm?

Well i don't know...
I just `sha256sum packager-osx-x64`.

I only did arm64 at first, because all the new Macs (since M1 2020) are equipped with arm chips. that was why I chose it.

File: Casks/s/shaka-packager.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  name "Shaka Packager"
+  desc "Packager for DASH and HLS streaming"
+  homepage "https://github.com/shaka-project/shaka-packager"
+
+  livecheck do
+    url :url
+    strategy :github_latest
+  end
+
+  binary "packager-osx-#{arch}", target: "packager"
```
Comment (SMillerDev) [inline Casks/s/shaka-packager.rb:18]: Please note the rejection reasons for casks:
> App is both open-source and CLI-only (i.e. it only uses the binary artifact). In that case, and [in the spirit of deduplication](https://github.com/Homebrew/homebrew-cask/issues/15603), submit it first to [homebrew/core](https://github.com/Homebrew/homebrew-core) as a formula that builds from source. If it is rejected, you may then try again as a cask (link to the issue from your pull request so we can see the discussion and reasoning for rejection).

https://docs.brew.sh/Acceptable-Casks#:~:text=App%20is%20both,reasoning%20for%20rejection).

File: Casks/s/shaka-packager.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+cask "shaka-packager" do
+  arch arm: "arm64", intel: "x64"
+
+  version "3.4.0"
+  sha256 arm:   "97acc637f83a3967e576612c2a627346f3a271bc018949c8e28b076b9cc517fc",
+         intel: "74f9236605042d364e55967e1679dcd6b7a9b3fdf6c66a40175090d27a3c24c6"
+
+  url "https://github.com/shaka-project/shaka-packager/releases/download/v#{version}/packager-osx-#{arch}"
```
Comment (SMillerDev) [inline Casks/s/shaka-packager.rb:8]: The x86 package does not meet the criteria for inclusion in Homebrew/cask:
> App fails with GateKeeper enabled on Homebrew supported macOS versions and platforms (e.g. unsigned apps fail on Macs with Apple Silicon/ARM).

https://docs.brew.sh/Acceptable-Casks#:~:text=App%20fails%20with%20GateKeeper%20enabled%20on%20Homebrew%20supported%20macOS%20versions%20and%20platforms%20(e.g.%20unsigned%20apps%20fail%20on%20Macs%20with%20Apple%20Silicon/ARM).

**Comment by khipp** [comment]
Thank you for your submission @garnajee, but we cannot accept it at this time for the reasons stated above.

Please submit this to [homebrew/core](https://github.com/Homebrew/homebrew-core) as a formula that builds from source. If it is rejected, please consider hosting it [in your own Tap](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap). If the developers sign their app in the future, we would be happy to consider adding it to `homebrew-cask`.

**Pull request closed without merging**